engine/camunda/src/main/java/org/camunda/bpm/delegate/InterfaceWfcHandler.java

```diff
@@ -181,13 +204,92 @@ public boolean createUserTask(DelegateTask delegateTask) {
      */
 
     private void requestData(String query, String processInstanceId, String messageName, String variableName,
-            String taskIdentifier, boolean isInterupting) {
+        String taskIdentifier, boolean isInterupting) {
         //// Building the complete URL for the request.
         String url = properties.getProperty("wfc.url") + "/RequestObservationValue/"
-                + processInstanceId + "/" + messageName + "/" + variableName + "/"
-                + taskIdentifier + "/" + isInterupting;
+            + processInstanceId + "/" + messageName + "/" + variableName + "/"
+            + taskIdentifier + "/" + isInterupting;
         // Sending the request to the WFC service and Unirest is used to send a POST
         // request to the WFC service.
         Unirest.post(url).body(query).asJson();
     }
+
+    /**
+     * This method processes FHIR query expressions containing time references like "NOW", "MOMENT", or custom variables.
+     * It handles parsing the base date/time, applying any specified offset, and formatting the final result.
+     *
+     * @param innerBaseValue The base date/time value from the expression (e.g., "NOW", "$VAR_NAME")
+     * @param innerOffset  The optional offset string (e.g., "5h", "-2m") containing a value and unit
+     * @param variableAccessor A way to access variables
+     * @param dateTimeFormatter The desired format for the final output date/time string
+     * @return The formatted date/time string after processing
+     */
+    private String HandleTimeExpression(String innerBaseValue, String innerOffset, VariableAccessor variableAccessor, DateTimeFormatter dateTimeFormatter) {
+
+        // Determine the base date/time
+        ZonedDateTime baseDateTime;
+        if ("NOW".equals(innerBaseValue) || "$NOW".equals(innerBaseValue)) {
```

Comment:
are &NOW and NOW both accepted ? Any reason ?

Reply:
Yes, both are accepted,
just to avoid confusion,
$(MOMENT($var-flag))
I can replace $var with NOW (considering NOW as a value)
also, can replace $var with $NOW (considering NOW as expression)

so, whichever is the choice above, the model will work.